src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -272,6 +312,12 @@ CHIP_ERROR DeviceControllerSystemState::Shutdown()
     mSystemLayer        = nullptr;
     mUDPEndPointManager = nullptr;
 
+    if (mCASEServer != nullptr)
+    {
+        chip::Platform::Delete(mCASEServer);
```

Comment:
Still wondering why this is after mExchangeMgr shutdown (though at least now it's before mExchangeMgr deletion)....  Seems weird.